### PR TITLE
Reddit processing code

### PR DESCRIPTION
**Pull request by drschwenk**
Adding the processing code for building five variations of Reddit pretraining data from the raw Pushshift comment and submission dumps. 

The code relies on apache beam and GCP's Dataflow service and so it sits apart from the rest of the Dolma codebase. It's included for reproducibility of the Reddit data in v1.5 (atomic_comments_v5) as well as the data used in the ablation experiments leading to that selection.

### CODE REVIEW

File: sources/reddit/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+A first experiment with atomic comments.
+
+## complete_threads_codelike_v4
+
+Only one variation of this version was tried. 
+
+## atomic_content_v5                       
+
+A refined version of atomic_content_v3, v5 uses different length and selection criteria for comments and submissions.
```
Comment (Muennighoff) [inline sources/reddit/README.md:40]: Is `atomic_content_v5` the final one to be included in Dolma?

File: sources/reddit/atomic_content_v5/build_comment_data.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        while len(text) > 1 and (text[-1].isalnum() == text[-2].isalnum()):
+            text = text[:-1]
+        return text[:-1]
+
+    comment = {
+        'id': comment['id'],
+        'thread_id': normalize_id(comment['link_id']),
+        'parent_id': normalize_id(comment['parent_id']),
+        'body': trim(normalize_string(comment['body']), max_length),
+        'body_is_trimmed': len(comment['body']) > max_length,
```
Comment (Muennighoff) [inline sources/reddit/atomic_content_v5/build_comment_data.py:70]: Why trim it if the trimmed ones seem to be filtered out anyways?

**Review by Muennighoff — COMMENTED** [review]
Hope you don't mind me commenting! This looks great! I'm computing the correlations of our different filters/taggers for each dataset. Am I understanding correctly that the filters for Reddit are:
- Length (Min & Max len)
- Score / Upvotes (Has to be >= 3)
- Over 18 (for submission data)

It seems like it's much less than we had for Common Crawl (which is fine with me; just wondering if a heatmap correlation of them will be very interesting cc @soldni)



Also on a high-level, I think it'd be nice to indicate which of the different variations is the "final one" i.e. the best one that I assume will be included in dolma? Maybe the others can just all be put into a separate folder named `alternatives` or something

File: sources/reddit/README.md (hunk 1 above)
Comment (soldni) [reply sources/reddit/README.md:40]: yes!

File: sources/reddit/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## atomic_content_v3
+
+A first experiment with atomic comments.
+
+## complete_threads_codelike_v4
+
+Only one variation of this version was tried. 
+
+## atomic_content_v5
```
Comment (soldni) [inline sources/reddit/README.md:38]:
```suggestion
## atomic_content_v5   (*included in Dolma v 1.5*)
```

**Review by soldni — APPROVED** [review]

**Pull request merged**